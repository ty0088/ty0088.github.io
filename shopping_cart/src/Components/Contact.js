import React from 'react';
import '../Style/style.css';

const Contact = () => {
    return (
        <div>
            <p>To contact our customer service team:</p>
            <br></br>
            <p>Email us: <a href="mailto:webmaster@example.com">cats-r-us@catsRusStore.cats</a></p>
            <br></br>
            <p>Call us: <a href="tel:012312341234">0123 1234 1234</a></p>
            <p>Our lines are open:</p>
            <p>Mon-Fri: 9am-5pm</p>
            <p>Sat-Sun: Closed</p> 
        </div>
    );
};

export default Contact;