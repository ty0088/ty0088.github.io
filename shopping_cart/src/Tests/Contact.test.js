import React from 'react';
import { render , screen } from '@testing-library/react';
import Contact from '../Components/Contact';

describe("Contact component", () => {
    it("Snapshot", () => {
      const { container } = render(<Contact />);
      expect(container).toMatchSnapshot();
    });

    it("Email link present", () => {
        render(<Contact />);
        expect(screen.getByRole("link", {name: "cats-r-us@catsRusStore.cats"})).toBeTruthy();
    });

    it("Phone link present", () => {
        render(<Contact />);
        expect(screen.getByRole("link", {name: "0123 1234 1234"})).toBeTruthy();
    });
});