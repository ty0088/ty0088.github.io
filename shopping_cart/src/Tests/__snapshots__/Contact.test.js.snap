// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Contact component Snapshot 1`] = `
<div>
  <div>
    <p>
      To contact our customer service team:
    </p>
    <br />
    <p>
      Email us: 
      <a
        href="mailto:webmaster@example.com"
      >
        cats-r-us@catsRusStore.cats
      </a>
    </p>
    <br />
    <p>
      Call us: 
      <a
        href="tel:012312341234"
      >
        0123 1234 1234
      </a>
    </p>
    <p>
      Our lines are open:
    </p>
    <p>
      Mon-Fri: 9am-5pm
    </p>
    <p>
      Sat-Sun: Closed
    </p>
  </div>
</div>
`;
